app/controllers/spree/api/products_controller_decorator.rb

```diff
@@ -33,7 +33,7 @@ def soft_delete
     authorize! :delete, Spree::Product
     @product = find_product(params[:product_id])
     authorize! :delete, @product
-    @product.delete
+    @product.destroy
     respond_with(@product, :status => 204)
   end
```

Comment:
can we create an issue to refactor this `#soft_delete`? Now that Spree has it we don't need a separate soft delete implementation. The frontend shouldn't care whether the delete is soft or not, IMO. It's an implementation detail.

Reply:
Done: #3405